gitlab/resource_gitlab_project.go

```diff
@@ -196,5 +197,16 @@ func resourceGitlabProjectDelete(d *schema.ResourceData, meta interface{}) error
 	log.Printf("[DEBUG] Delete gitlab project %s", d.Id())
 
 	_, err := client.Projects.DeleteProject(d.Id())
-	return err
+	if err != nil {
+		return err
+	}
+	for {
+		_, response, err := client.Projects.GetProject(d.Id())
+		if err != nil {
+			if response.StatusCode == 404 {
+				return nil
+			}
+		}
+		time.Sleep(time.Second * 1)
```

Comment:
it'd be good to add a max attempts timeout here, if possible? i.e. try 30 times then raise an error?

Reply:
I was relying on terraform to do this, e.g. time out?

Reply:
isnt that a terraform core functionnality?

Reply:
Terraform exposes this as the `context` but I thought it needed to be threaded through? (I may be wrong!)

In general we'd [tend to achieve this using `StateChangeConf`](https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/resource_vsphere_datacenter.go#L64) to ensure it's deleted

Reply:
I have implemented that. Waiting for your review to merge.